### ISSUE TITLE
The handling of nil byte slices is different than the handling of other types of nil slices

### ISSUE THREAD

**Issue by aboodman** [polydawn/refmt#27]
```
type T struct{}
var t1, t2 []T
data, _ = cbor.Marshal(t1)
cbor.Unmarshal(data, &t2)
assert.Nil(t1)  // true
assert.Nil(t2)  // true

var b1, b2 []byte
data, _ := cbor.Marshal(b1)
cbor.Unmarshal(data, &b2)
assert.Nil(b1) // true
assert.Nil(b2) // false
```

**Comment by aboodman** [comment]
Note: I actually prefer what it's doing for byte slices for my use case, or alternately making it optional as suggested in #25.

**Comment by warpfork** [comment]
Fixed by merging #28 (a while ago).
